crates/e2e/tests/e2e/smart_contract_orders.rs

```diff
@@ -262,5 +305,31 @@ async fn smart_contract_orders(web3: Web3) {
         .call()
         .await
         .expect("Couldn't fetch native token balance");
-    assert_eq!(balance, U256::from(8_972_194_924_949_384_291_u128));
+    assert_eq!(balance, U256::from(7_975_363_884_976_534_272_u128));
+}
+
+async fn gnosis_safe_eip1271_signature(
+    key: SecretKeyRef<'_>,
+    safe: &GnosisSafe,
+    domain: &DomainSeparator,
+    order: OrderBuilder,
+) -> Vec<u8> {
+    let handler =
+        GnosisSafeCompatibilityFallbackHandler::at(&safe.raw_instance().web3(), safe.address());
+
+    let order_hash = hashed_eip712_message(domain, &order.build().data.hash_struct());
+    let message_hash = handler
+        .get_message_hash(Bytes(order_hash.to_vec()))
+        .call()
+        .await
+        .unwrap();
+
+    let signature = key.sign(&message_hash.0, None).unwrap();
+
+    let mut bytes = vec![0u8; 65];
+    bytes[0..32].copy_from_slice(signature.r.as_bytes());
+    bytes[32..64].copy_from_slice(signature.s.as_bytes());
+    bytes[64] = signature.v as _;
```

Comment:
Does this need to be initialized the same way [this](https://github.com/cowprotocol/services/pull/315/files#diff-971b3514de01c1b5a904a01cfe9f58f8ec2caede1def99d50971a2c79e5ec638R90-R92) was?